test/test-api.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// Testcase to produce NodeReport via API call
+if (process.argv[2] === 'child') {
+  const nodereport = require('../');
+  nodereport.triggerReport();
+} else {
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+  const tap = require('tap');
+
+  const child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', (code) => {
+    tap.plan(3);
+    tap.equal(code, 0, 'Process exited cleanly');
+    // Locate and validate the NodeReport
```

Comment:
I wouldn't have this comment. Comment says

    Locate and validate the NodeReport

code says

    findReports ... validate(... report )...

so comment is just rewording the code. Just my thoughts.